SearchParser.js

```diff
@@ -0,0 +1,49 @@
+/**
+ * SearchParser
+ * @param {Array} patterns  Enumerates params and the patterns that match them
+ * @param {boolean} options.returnAllMatches  If true return all matches (not just first)
```

Comment:
Oh, that makes sense. I don't think there's any need for `returnAllMatches` in the current quick search use cases, so I'll remove it.